### ISSUE TITLE
Exporting/Importing Bookmarks

### ISSUE THREAD

**Issue by user-64bit** [medyo/hackertab.dev#156]
I've been using Hackertab from past few weeks. I'm using it in my work laptop as well as my personal laptop.
So the issue happens when I try to get the bookmarked article from one laptop to another one. or wants to share the bookmarks with someone.

Exporting/Importing bookmarks will help anyone to organize there favorite articles.

Hope you got the point :)

**Comment by idebenone** [comment]
Hey @medyo !
Can I work on this feature?

**Comment by medyo** [comment]
@idebenone with pleasure,
Can you provide a compact tech spec for the feature, explain your proposed implementation method, and share any available UIs if available.

**Comment by idebenone** [comment]
Hey @medyo !
I was thinking of using storage.sync which is a part of Chrome/Firefox extension API and have an option for automatic/manual sync between devices.

**Comment by idebenone** [comment]
Update !
I tried to do with **storage.sync** , but its not reliable. IMO it would be better to do it from backend(ex: OAuth) and plus we can save a lot metadata like UI state and user preferences which will be in sync across devices. I think backend is not available to work on as of now ryt?
Please let me know what you think of this @medyo.

**Comment by arth2002** [comment]
Hello @idebenone
You might be right. If will be better if we can implement it from the back end.
But I guess for now we can at least provide a feature of Import/Export.
We can Optimize or make it better later on.
Tell me, If I'm missing something here?

What do you say? @medyo

**Comment by medyo** [comment]
@arth2002 @idebenone did you check the [`chrome.bookmarks`](https://developer.chrome.com/docs/extensions/reference/bookmarks/) API?

**Comment by idebenone** [comment]
@medyo 
Interesting! Will look into it.

**Comment by idebenone** [comment]
Hi @medyo,
I wanted to work on this feature, but apparently I'm getting an error. Tried re-installing the packages, still no luck.

![image](https://github.com/medyo/hackertab.dev/assets/114807970/52492aec-6cd1-4a9b-b873-b7db68699f90)

**Comment by medyo** [comment]
> Hi @medyo, I wanted to work on this feature, but apparently I'm getting an error. Tried re-installing the packages, still no luck.
> 
> ![image](https://private-user-images.githubusercontent.com/114807970/291188434-52492aec-6cd1-4a9b-b873-b7db68699f90.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4OTY1NTksIm5iZiI6MTcwMjg5NjI1OSwicGF0aCI6Ii8xMTQ4MDc5NzAvMjkxMTg4NDM0LTUyNDkyYWVjLTZjZDEtNGE5Yi1iODczLWI3ZGI2ODY5OWY5MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxOFQxMDQ0MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MzQ1YTUzNjkzZGU1Y2JiNDVlYzllZTUzYjQ5NzAwMTcxYTE2MDgwYThkMDQ5OTUyMGJjMDUyNmNmZDEwZjRiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7xkrXxjK3uo8FWbMARkPY0SykaFvc2T55SycWeFjkJI)

I think `react-icons` has removed or changed the icon name, please replace in the project `GoPrimitiveDot` by `GoDotFill` and push the fix (to `develop`)

**Comment by medyo** [comment]
@idebenone Fixed, feel free to pull the last changes

**Comment by idebenone** [comment]
@medyo 
I have created a function which will create a folder(bookmark's folder) in browser and adds all the hackertab bookmarks. Is this sufficient?

**Comment by medyo** [comment]
yes sounds good, do you ask for the bookmarks permission on click ?

**Comment by idebenone** [comment]
Not at the moment, but I will include if that's required.

**Comment by us3r64bit** [comment]
Hey @idebenone 
Are you working on this issue? 👀
If not then I would like to do it myself if possible. 🙂
Thanks 😊 
cc: @medyo

**Comment by idebenone** [comment]
Hey @us3r64bit 
Yeah sure, my hands are full at the moment.

**Comment by us3r64bit** [comment]
Hello @medyo 
It might be so stupid to ask for this type of doubt but I'm very bad with setting up projects on local. :sweat_smile: 
I was getting error and I don't know WTH is `VITE_API_URL` and where to get it from. (I've never used VITE too :face_with_spiral_eyes:)
Please guide me about how can I get `VITE_API_URL` and other env variables if there are any.
Thanks :slightly_smiling_face:

**Comment by medyo** [comment]
Hi @us3r64bit, Actually I forgot to update the README,

create an .env at the root of the project with the following vars:
```
VITE_BUILD_TARGET=extension
VITE_API_URL=https://api.hackertab.dev/
```

**Comment by us3r64bit** [comment]
Pinging you in case you miss my message on PR. :slightly_smiling_face: 
@medyo

**Comment by medyo** [comment]
The feature has been deployed, I forgot to update you
